src/Avif Reader/AvifReaderImage.cs

```diff
@@ -0,0 +1,277 @@
+                temp?.Dispose();
+            }
+        }
+
+        void IOutputImageTransform.Rotate180()
+        {
+            using (IBitmapLock bitmapLock = this.image.Lock(BitmapLockOptions.ReadWrite))
+            {
+                PixelFormat pixelFormat = bitmapLock.PixelFormat;
+
+                if (pixelFormat == PixelFormats.Bgra32)
+                {
+                    ImageTransform.Rotate180<ColorBgra32>(bitmapLock);
+                }
+                else if (pixelFormat == PixelFormats.Rgb64)
+                {
+                    ImageTransform.Rotate180<ColorRgba64>(bitmapLock);
+                }
+                else if (pixelFormat == PixelFormats.Rgba128Float)
+                {
+                    ImageTransform.Rotate180<ColorRgba128Float>(bitmapLock);
+                }
+                else
+                {
+                    ExceptionUtil.UnsupportedPixelFormat(pixelFormat);
+                }
+            }
+        }
+
+        void IOutputImageTransform.Rotate270CCW()
+        {
+            IBitmap? temp = null;
+
+            try
+            {
+                SizeInt32 size = this.image.Size;
+                PixelFormat pixelFormat = this.image.PixelFormat;
+
+                temp = this.imagingFactory.CreateBitmap(size.Height, size.Width, pixelFormat);
+
+                using (IBitmapLock sourceLock = this.image.Lock(BitmapLockOptions.Read))
+                using (IBitmapLock destLock = temp.Lock(BitmapLockOptions.Write))
+                {
+                    if (pixelFormat == PixelFormats.Bgra32)
+                    {
+                        ImageTransform.Rotate270CCW<ColorBgra32>(sourceLock, destLock);
+                    }
+                    else if (pixelFormat == PixelFormats.Rgb64)
+                    {
+                        ImageTransform.Rotate270CCW<ColorRgba64>(sourceLock, destLock);
+                    }
+                    else if (pixelFormat == PixelFormats.Rgba128Float)
+                    {
+                        ImageTransform.Rotate270CCW<ColorRgba128Float>(sourceLock, destLock);
+                    }
+                    else
+                    {
+                        ExceptionUtil.UnsupportedPixelFormat(pixelFormat);
+                    }
+                }
+
+                this.image.Dispose();
+                this.image = temp;
+                temp = null;
+            }
+            finally
+            {
+                temp?.Dispose();
+            }
+        }
+
+        void IOutputImageTransform.FlipHorizontal()
+        {
+            using (IBitmapLock bitmapLock = this.image.Lock(BitmapLockOptions.ReadWrite))
+            {
+                PixelFormat pixelFormat = bitmapLock.PixelFormat;
+
+                if (pixelFormat == PixelFormats.Bgra32)
+                {
+                    ImageTransform.FlipHorizontal<ColorBgra32>(bitmapLock);
+                }
+                else if (pixelFormat == PixelFormats.Rgb64)
```

Comment:
Do you actually mean `Rgb64` or is this a typo and you actually meant `Rgba64`?

Reply:
Fixed.